src/main/scala/com/ing/wbaa/gargoyle/proxy/provider/LineageProviderAtlas.scala

```diff
@@ -0,0 +1,147 @@
+      outputType: String,
+      contentType: ContentType) = {
+    Entities(Seq(IngestedFile(
+      "DataFile",
+      userSTS,
+      FileAttributes(bucketObject, bucketObject, bucketObject, contentType.mediaType.value,
+        guidRef(bucketGuid, "Bucket"),
+        guidRef(serverGuid, "Server"),
+        List(guidRef(inputGuid, inputType)),
+        List(guidRef(outputGuid, outputType)),
+        Seq(Classification("customer_PII"))))))
+  }
+
+  private def processEntities(
+      serverGuid: String,
+      bucketGuid: String,
+      fileGuid: String,
+      userSTS: String,
+      method: String,
+      inputGuid: String,
+      inputType: String,
+      outputGuid: String,
+      outputType: String,
+      timestamp: Long) = {
+    Entities(Seq(Ingestion(
+      "aws_cli_script",
+      userSTS,
+      IngestionAttributes(s"aws_cli_${timestamp}", s"aws_cli_${timestamp}", method, userSTS,
+        guidRef(serverGuid, "Server"),
+        List(guidRef(inputGuid, inputType)),
+        List(guidRef(outputGuid, outputType))))))
+  }
+
+  private def postEnities(userSTS: String, host: String, bucket: String, bucketObject: String, method: String, contentType: ContentType, timestamp: Long): Future[LineagePostGuidResponse] = {
+    for {
+      serverGuid <- postData(serverEntities(userSTS, host).toJson)
+      bucketGuid <- postData(bucketEntities(userSTS, bucket).toJson)
+      fileGuid <- method match {
+        case access if access == "read" =>
+          postData(
+            fileEntities(serverGuid, bucketGuid, bucketObject, userSTS, bucketGuid, "Bucket", serverGuid, "Server", contentType).toJson)
+        case access if access == "write" =>
+          postData(
+            fileEntities(serverGuid, bucketGuid, bucketObject, userSTS, serverGuid, "Server", bucketGuid, "Bucket", contentType).toJson)
+      }
+      processGuid <- method match {
+        case access if access == "read" =>
+          postData(
+            processEntities(serverGuid, bucketGuid, fileGuid, userSTS, method, bucketGuid, "Bucket", fileGuid, "DataFile", timestamp).toJson)
+        case access if access == "write" =>
+          postData(
+            processEntities(serverGuid, bucketGuid, fileGuid, userSTS, method, fileGuid, "DataFile", bucketGuid, "Bucket", timestamp).toJson)
+      }
+    } yield LineagePostGuidResponse(serverGuid, bucketGuid, fileGuid, processGuid)
+  }
+
+  // file entity delete
+  // for now it is just deleting file entity and no related objects like eg. aws_cli_script, which uploaded or downloaded
+  // file to bucket. Once we delete aws_cli_script object we will lose track of whats has been deleted
+  // We need to come up with process of tracking file delete
+  private def deleteEntities(typeName: String, entityName: String): Future[LineagePostGuidResponse] =
+    for {
+      entityGuid <- getEntityGUID(typeName, entityName)
+      _ <- deleteEntity(entityGuid)
+
+    } yield LineagePostGuidResponse("", "", entityGuid, "")
```

Comment:
Instead of empty strings I think it's nicer to make these values optional in the case class and set them to None here